src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/ConsensusRuleUnitTestBase.cs

```diff
@@ -56,20 +57,27 @@ protected ConsensusRuleUnitTestBase(Network network)
             };
         }
 
-        protected static void AddBlocksToChain(ConcurrentChain chain, int blockAmount)
+        protected void AddBlocksToChain(ConcurrentChain chain, int blockAmount)
         {
             var nonce = RandomUtils.GetUInt32();
             var prevBlockHash = chain.Tip.HashBlock;
+
+            this.ruleContext.Set = new UnspentOutputSet();
```

Comment:
by the way, these preparations were for the UpdateCoinViewRule that I had started writing using the unit test base that neurosploit put in.

However, that didn't end up making it into this PR as I couldn't quite get it working - however it still seemed reasonable to leave this coinview setup inside the base as it didn't have one previously and whilst not strictly needed for the tests in place now, it is still a realistic setup of the engine - So I kept the code here breaking YAGNI.  

If you think I should undo this file, I am happy to.